bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -4379,6 +4490,627 @@ void ReplicaImp::setConflictDetectionBlockId(const ClientRequestMsg &clientReqMs
+  IRequestsHandler::ExecutionRequestsQueue *pAccumulatedRequests = new IRequestsHandler::ExecutionRequestsQueue;
+  size_t reqIdx = 0;
+  RequestsIterator reqIter(ppMsg);
+  char *requestBody = nullptr;
+  while (reqIter.getAndGoToNext(requestBody)) {
+    size_t tmp = reqIdx;
+    reqIdx++;
+    ClientRequestMsg req((ClientRequestMsgHeader *)requestBody);
+
+    if (!requestSet.get(tmp) || req.requestLength() == 0) {
+      clientsManager->removePendingForExecutionRequest(req.clientProxyId(), req.requestSeqNum());
+      continue;
+    }
+
+    //    SCOPED_MDC_CID(req.getCid());
+    NodeIdType clientId = req.clientProxyId();
+
+    pAccumulatedRequests->push_back(IRequestsHandler::ExecutionRequest{
+        clientId,
+        static_cast<uint64_t>(lastExecutedSeqNum + 1),
+        ppMsg->getCid(),
+        req.flags(),
+        req.requestLength(),
+        req.requestBuf(),
+        std::string(req.requestSignature(), req.requestSignatureLength()),
+        static_cast<uint32_t>(config_.getmaxReplyMessageSize() - sizeof(ClientReplyMsgHeader)),
+        (char *)std::malloc(config_.getmaxReplyMessageSize() - sizeof(ClientReplyMsgHeader)),
+        req.requestSeqNum()});
+
+    if (req.flags() & HAS_PRE_PROCESSED_FLAG) {
+      setConflictDetectionBlockId(req, pAccumulatedRequests->back());
+    }
+  }
+  if (ReplicaConfig::instance().blockAccumulation) {
+    LOG_DEBUG(GL,
+              "Executing all the requests of preprepare message with cid: " << ppMsg->getCid() << " with accumulation");
+    {
+      //      TimeRecorder scoped_timer1(*histograms_.executeWriteRequest);
+      const concordUtils::SpanContext &span_context{""};
+      auto span = concordUtils::startChildSpanFromContext(span_context, "bft_client_request");
+      span.setTag("rid", config_.getreplicaId());
+      span.setTag("cid", ppMsg->getCid());
+      span.setTag("seq_num", ppMsg->seqNumber());
+      bftRequestsHandler_->execute(*pAccumulatedRequests, time, ppMsg->getCid(), span);
+    }
+  } else {
+    LOG_INFO(
+        GL,
+        "Executing all the requests of preprepare message with cid: " << ppMsg->getCid() << " without accumulation");
+    IRequestsHandler::ExecutionRequestsQueue singleRequest;
+    for (auto &req : *pAccumulatedRequests) {
+      singleRequest.push_back(req);
+      {
+        //        TimeRecorder scoped_timer1(*histograms_.executeWriteRequest);
+        const concordUtils::SpanContext &span_context{""};
+        auto span = concordUtils::startChildSpanFromContext(span_context, "bft_client_request");
+        span.setTag("rid", config_.getreplicaId());
+        span.setTag("cid", ppMsg->getCid());
+        span.setTag("seq_num", ppMsg->seqNumber());
+        bftRequestsHandler_->execute(singleRequest, time, ppMsg->getCid(), span);
+        if (config_.timeServiceEnabled) {
+          time.request_position++;
+        }
+      }
+      req = singleRequest.at(0);
+      singleRequest.clear();
+    }
```

Comment:
Too big 'ifs'

Reply:
Just copied the current code, will do better refactoring in the next PR's.